src/pymodaq/daq_utils/h5backend.py

```diff
@@ -0,0 +1,892 @@
+    logger.warning(str(e))
+    is_tables = False
+
+is_h5py = True
+# other possibility
+try:
+    import h5py
+    backends_available.append('h5py')
+except Exception as e:                              # pragma: no cover
+    logger.warning(str(e))
+    is_h5y = False
+
+is_h5pyd = True
+# this one is to be used for remote reading/writing towards a HSDS server (or h5serv), see HDFGroup
+try:
+    import h5pyd
+    backends_available.append('h5pyd')
+except Exception as e:                              # pragma: no cover
+    logger.warning(str(e))
+    is_h5yd = False
+
+if not (is_tables or is_h5py or is_h5pyd):
+    logger.exception('No valid hdf5 backend has been installed, please install either pytables or h5py')
+
+#As I understand, the Node object should never be instanciated
+class Node(object):
+    def __init__(self, node, backend):
+        if isinstance(node, Node):  # to ovoid recursion if one call Node(Node()) or even more
+            self._node = node.node
+        else:
+            self._node = node
+        self.backend = backend
+        self._attrs = Attributes(self, backend)
+
+    def __str__(self):
+        # Get this class name
+        classname = self.__class__.__name__
+        # The title
+        title = self.attrs['TITLE']
+        return "%s (%s) %r" % \
+               (self.path, classname, title)
+
+    @property
+    def node(self):
+        return self._node
+
+    def __eq__(self, other):
+        return self.node == other.node
+
+    @property
+    def parent_node(self):
+        if self.path == '/':
+            return None
+        mod = importlib.import_module('.h5backend', 'pymodaq.daq_utils')
+
+        if self.backend == 'tables':
+            p = self.node._v_parent
+        else:
+            p = self.node.parent
+        klass = get_attr(p, 'CLASS', self.backend)
+        _cls = getattr(mod, klass)
+        return _cls(p, self.backend)
+
+    def set_attr(self, key, value):
+        self.attrs[key] = value
+
+    def get_attr(self, item):
+        return self.attrs[item]
+
+    @property
+    def attrs(self):
+        return self._attrs
+
+    @property
+    def name(self):
+        """return node name
+        """
+        if self.backend == 'tables':
+            return self._node._v_name
+        else:
+            path = self._node.name
+            if path == '/':
+                return path
+            else:
+                return path.split('/')[-1]
+
+    @property
+    def path(self):
+        """return node path
+        Parameters
+        ----------
+        node (str or node instance), see h5py and pytables documentation on nodes
+
+        Returns
+        -------
+        str : full path of the node
+        """
+        if self.backend == 'tables':
+            return self._node._v_pathname
+        else:
+            return self._node.name
+
+    def get_file(self):
+        """ Return node file. If this is a
```

Comment:
create a File Wrapper that would be returned from this method. In the File wrapper, at least the method copy should be included for all backends.

Reply:
Clearly it would be the responsibility from the h5backend object, which mostly encapsulates the file object, no?

Reply:
I thought about this and looked int the h5 backend. I think it would be good to remove the few things there that are backend specifics and move them into their respective object. For instance: in the H5Backend object there is a method called `get_children`, this one is typically belonging to a GROUP object and indeed when looking into it i saw I created the children method that has exactly the same content... So in the backend the content of the method could be as simply as:

```
    def get_children(self, where):
        """Get a dict containing all children node hanging from where with their name as keys and types among Node,
        """
        where = self.get_node(where)  # return a node object in case where is a string
        if isinstance(where, GROUP):
             return where.children()
        else:
             return dict([])
``` 
I don't know why I missed that. But checking all backend specifics from H5Backend and moving them into their dedicated objects (Node, GROUP, File (to be created)) would make the code clearer and easier to maintain. Then for our initial purpose, the node group would need a get_file returning a File object itself having a copy method. So it could be called from any node.

I think we should end here the review and accept the merge and it would be great, because you dived into this backend stuff, to apply this in another pull request (if you're willing) (or at least to open an issue, copying this comment content as a starting point)!
With pymodaq you know where you start but you don't know where it will lead you ;-)

Reply:
Let me know if you're ok with merging now?

Reply:
Definitely if you can handle these things from the Node sub classes directly it's better. I would go as far as saying that some methods could be moved outside classes altogether.